app/gameengine.py

```diff
@@ -2164,6 +2177,67 @@ def continue_perform_art(self):
         }
         self.deal_damage(active_player, target_owner, self.performance_performer_card, self.performance_target_card, total_power, is_special_damage, False, art_info, self.performance_continuation)
 
+    def deal_life_damage(self, target_player: PlayerState, dealing_card, damage: int, continuation):
+        game_over = False
+        game_over_reason = ""
+        life_to_distribute = []
+        life_lost = damage
+
+        current_life = len(target_player.life)
+        if life_lost >= current_life:
+            game_over = True
+            game_over_reason = GameOverReason.GameOverReason_NoLifeLeft
+
+        if not game_over:
+            life_to_distribute = ids_from_cards(target_player.life[:life_lost])
+
+        # Send the life damage dealt event
+        life_damage_event = {
+            "event_type": EventType.EventType_LifeDamageDealt,
+            "target_player": target_player.player_id,
+            "source_card_id": dealing_card["game_card_id"],
+            "life_lost": life_lost
+        }
+        self.broadcast_event(life_damage_event)
+
+        if game_over:
```

Comment:
if all of this is identical (or nearly) to the other location where this happens, this should be a function.
maybe process_life_lost or something along those lines?